keras_tuner/engine/tuner_utils.py

```diff
@@ -201,6 +206,10 @@ def on_epoch_end(self, epoch, logs=None):
         if self.objective.better_than(current_value, self.best_value):
             self.best_value = current_value
             self.model.save_weights(self.filepath)
+            # Remove temporary saved model files from non-chief workers.
+            ds_utils.remove_temp_dir_with_filepath(
+                self.filepath, self.model.distribute_strategy
+            )
```

Comment:
Two questions about this part:
1. Why do we only remove the temp dir when we got a better result? Do we do it only when saving weights? However we also did a save_weights on line 203.
2. In the doc of `remove_temp_dir_with_filepath()`, the first arg should be the original file path, but it seems to be using the temporary file path assigned on line 197.